src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -622,6 +631,154 @@ private void LogExceptionEvents(List<ExecutionException> exceptions, CompilerDat
+                        {
+                            sb.AppendLine(
+                                $"`{CompilerDataLogger.AssemblyReferencesEventName}` event detected with unexpected Id. " +
+                                $"Expected `{assemblyEventsId}` but found `{currentAssemblyEventId}`.");
+                        }
+
+                        break;
+                    }
+                    case CompilerDataLogger.CommandLineEventName:
+                    {
+                        commandLineEvents.Add(telemetryEvent);
+
+                        if (expectedEventName == CompilerDataLogger.CommandLineEventName
+                            && !expectedChunkedContent.Contains(properties["chunkedcommandLine"]))
+                        {
+                            sb.AppendLine(
+                                $"Unexpected `{CompilerDataLogger.CommandLineEventName}` chunked content: " +
+                                $"`{properties[CompilerDataLogger.ChunkedCommandLine]}` " +
+                                $"expected: `{expectedChunkedContent}`");
+                        }
+
+                        string currentCommandLineEventId = properties["commandLineId"];
+
+                        if (commandLineEventsId != currentCommandLineEventId)
+                        {
+                            sb.AppendLine(
+                                $"`{CompilerDataLogger.CommandLineEventName}` event detected with unexpected Id. " +
+                                $"Expected `{commandLineEventsId}` but found `{currentCommandLineEventId}`.");
+                        }
+                        break;
+                    }
+                    case CompilerDataLogger.SummaryEventName:
+                    {
+                        summaryEvents.Add(telemetryEvent);
+                        break;
+                    }
+                }
+            }
+
+            if ((expectedEventName == CompilerDataLogger.AssemblyReferencesEventName
+                    && assemblyReferencesEvents.Count != chunkCount) ||
+                (expectedEventName == CompilerDataLogger.CommandLineEventName
+                    && commandLineEvents.Count != chunkCount))
+            {
+                sb.AppendLine(
+                    $"Expected `{chunkCount}` `{expectedEventName}` events, " +
+                    $"but `{assemblyReferencesEvents.Count}` were found");
+            }
+
+            return sb;
+        }
+
+        private StringBuilder ValidateChunkedContent(StringBuilder sb,
+                                                     int expectedChunkSize,
+                                                     List<EventTelemetry> chunkedEvents)
+        {
+            if (chunkedEvents.Count != expectedChunkSize)
+            {
+                sb.AppendLine(
+                    string.Format(
+                        "Incorrect number of chunkedEvents de tected. Expected {0}, but found {1}",
```

Comment:
>de tected. [](http://example.com/codeflow?start=59&length=10)

Is there an erroneous space in 'de tected'? :) You can use interpolated strings here as well.

I'm approving this change but if you have a moment to go fix these nits, do that. :)